### ISSUE TITLE
DatePicker: JavaScript error during ajax request

### ISSUE THREAD

**Issue by JamieGHamilton** [primefaces/primefaces#6641]
**Describe the defect**
The datePicker widget javascript contains the method "_destroy" which gets called during an ajax request when the component is rerendered.  This destroy method calls 2 methods: this.restoreOverlayAppend() and this.onOverlayHide() - neither function seems to be defined anywhere and creates javascript errors.

**Reproducer**

**Environment:**
 - PF Version: 8.0
 - JSF + version: MyFaces 2.3.6
 - Affected browsers: ALL

**To Reproduce**
1. Setup page with example xhtml below
2. Monitor for javascript errors
3. Click button "click for javascript error"
4. See error: "this.restoreOverlayAppend is not a function"

**Expected behavior**
The datePicker destroy code should succeed without javascript errors.

**Example XHTML**
```xml
<h:form>
	<p:datePicker id="dp" />

	<p:commandButton id="ajaxButton"
	                 value="click for javascript error"
	                 process="dp"
	                 update="dp" />
</h:form>
```
**Example Bean**
no bean required

**Comment by melloware** [comment]
I can't reproduce?
[pf-6641.zip](https://github.com/primefaces/primefaces/files/5675079/pf-6641.zip)

Run `mvn clean jetty:run -Pmyfaces23` and navigate to http://localhost:8080/primefaces-test/test.xhtml

**Comment by JamieGHamilton** [comment]
MyFaces will capture and hide this exception so you have to turn on debugging and pause on caught exceptions.
I can reproduce it with your example:
![image](https://user-images.githubusercontent.com/24283616/101818884-8272f280-3af2-11eb-90d2-e51a49390ded.png)

**Comment by melloware** [comment]
I have my Chrome Console set to "preserve log" so any error thrown in the console remains and does not get cleared and I don't see it?
![image](https://user-images.githubusercontent.com/4399574/101819260-fad9b380-3af2-11eb-89d9-7b91e1914eda.png)

**Comment by JamieGHamilton** [comment]
In chrome you need the stop sign to be blue and the checkbox "Pause on caught exceptions"
![image](https://user-images.githubusercontent.com/24283616/101819482-5441e280-3af3-11eb-8706-0182a9bbf4ff.png)

**Comment by melloware** [comment]
I see it now thanks!

**Comment by melloware** [comment]
OK can you add this MonkeyPatch to your JS code that loads after PF and see if its fixed.
```javascript
if ($.prime && $.prime.datePicker) {
    $.prime.datePicker.prototype._destroy = function() {
        this.hideOverlay();
    };
}
```

**Comment by JamieGHamilton** [comment]
Yes...it's fixed!

**Comment by melloware** [comment]
Thanks for reporting I would have never seen this if not for that break on JS error checkbox!

**Comment by JamieGHamilton** [comment]
No problem.

**Comment by tibor-both-cc** [comment]
Can the fix patched also to the 8.x.x versions?

**Comment by melloware** [comment]
Most likely not in 8.0 as the fix is already included in 10.0.0 and 11.0.0 and you have the MonkeyPatch you can use above for 8.0
